### PR TITLE
feature: add calculate_capacity for output and script

### PR DESCRIPTION
**Pull request by u2**

### CODE REVIEW

**Comment by ashchan** [comment]
Can we hold this until we release `v0.12.0` so that we can spare no effect on release that version?

After `rc/v0.12.0` freeze this can be merged to `develop`.

File: lib/ckb/types/script.rb [hunk 1]
```diff
@@ -12,6 +12,13 @@ def initialize(code_hash:, args:)
         @args = args
       end
 
+      # @return [Integer]
+      def calculate_capacity
+        capacity = 1 + (@args || []).map(&:bytesize).reduce(0, &:+)
```
Comment (doitian) [inline lib/ckb/types/script.rb]: Where does the `1` come from?

Comment (doitian) [reply lib/ckb/types/script.rb]: This function is wrong to add 1.

File: lib/ckb/types/script.rb [hunk 2]
```diff
@@ -12,6 +12,13 @@ def initialize(code_hash:, args:)
...
+        capacity += Utils.hex_to_bin(@code_hash).bytesize if @code_hash
```
Comment (doitian) [inline lib/ckb/types/script.rb]: The name is confusing and inconsistent. Capacity should always use Shannon as the uint.

File: lib/ckb/types/script.rb (hunk 1 above)
Comment (u2) [reply lib/ckb/types/script.rb]: fixed

**Comment by u2** [comment]
updated

File: lib/ckb/types/script.rb (hunk 2 above)
Comment (u2) [reply lib/ckb/types/script.rb]: fixed

**Comment by doitian** [comment]
We can add a guard in `send_capacity` to check whether the capacity is enough to hold the output.

**Pull request merged**